x-pack/plugins/security/server/lib/authorization/privileges.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Feature } from '../../../../xpack_main/types';
+import { IGNORED_TYPES } from '../../../common/constants';
+import { Actions } from './actions';
+import { FeaturesPrivileges, FeaturesPrivilegesBuilder } from './features_privileges_builder';
+
+export interface PrivilegeMap {
+  global: Record<string, string[]>;
+  features: FeaturesPrivileges;
+  space: Record<string, string[]>;
+}
+
+export interface PrivilegesService {
+  get(): PrivilegeMap;
+}
+
+interface XPackMainPlugin {
+  getFeatures(): Feature[];
+}
+
+export function privilegesFactory(
+  allSavedObjectTypes: string[],
+  actions: Actions,
+  xpackMainPlugin: XPackMainPlugin
+) {
+  return {
+    get() {
+      // TODO: I'd like to ensure an explicit Error is thrown here if all
```

Comment:
I like this idea. Do you know how we'd be able to detect this?

Reply:
Spencer said there was "a way" but he was in the middle of triaging a 6.5 upgrade issue, so I'll be following up with him on this today. I think we should probably add the check in the `getFeatures` implementation itself though...